src/preact/component/intersection-observer-resize.js

```diff
@@ -0,0 +1,37 @@
+import {
+  observeWithSharedInOb,
+  unobserveWithSharedInOb,
+} from '#core/dom/layout/viewport-observer';
+
+import {useCallback, useRef} from '#preact';
+
+/**
+ * Uses a shared IntersectionObserver per window instance to observe the given `ref`.
+ *
+ * @param {function(IntersectionObserverEntry)} callback
+ * @return {function}
+ */
+export function useIntersectionObserver(callback) {
+  const nodeRef = useRef(null);
+  const refCb = useCallback(
+    (node) => {
+      const prevNode = nodeRef.current;
+      nodeRef.current = node;
+      if (prevNode) {
+        unobserveWithSharedInOb(prevNode);
+      }
+      if (!node) {
+        return;
+      }
+
+      observeWithSharedInOb(node, callback);
+
+      return () => {
+        unobserveWithSharedInOb(node);
+      };
```

Comment:
This is being used as a ref fn, so there's no return value. Cleanup has to happen during the null call, which we're already taking care of above.

```suggestion
```